src/Transport/Curl.php

```diff
@@ -206,7 +206,14 @@ public function request($url, $headers = [], $data = [], $options = []) {
 
 		$options['hooks']->dispatch('curl.after_send', []);
 
-		if (curl_errno($this->handle) === CURLE_WRITE_ERROR || curl_errno($this->handle) === CURLE_BAD_CONTENT_ENCODING) {
+		$curl_errno = curl_errno($this->handle);
+
+		if ($curl_errno === CURLE_WRITE_ERROR && $this->response_byte_limit && $this->response_byte_limit === $this->response_bytes) {
```

Comment:
```suggestion
		if ($curl_errno === CURLE_WRITE_ERROR && $this->response_byte_limit && $this->response_bytes >= $this->response_byte_limit) {
```

Just to be on the safe side ?